pkg/workflow/operation/operation.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operation
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/common"
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	"github.com/oam-dev/kubevela/pkg/apiserver/domain/service"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application"
+	"github.com/oam-dev/kubevela/pkg/controller/utils"
+	"github.com/oam-dev/kubevela/pkg/oam"
+	"github.com/oam-dev/kubevela/pkg/resourcetracker"
+	"github.com/oam-dev/kubevela/pkg/rollout"
+	errors3 "github.com/oam-dev/kubevela/pkg/utils/errors"
+
+	"github.com/pkg/errors"
+)
+
+// WfOperator is opratior handler for workflow's resume/rollback/restart
+type WfOperator interface {
```

Comment:
It is cool to move the actions of Workflow into a separate package. Could it be renamed as WorkflowOperator to make it clear?

Reply:
If possible, I would prefer an alternative usage like
`NewApplicationClient(k8sClient).Workflow().Suspend(ctx, app)` but it is okay for the WorkflowOperator as well.

Reply:
> It is cool to move the actions of Workflow into a separate package. Could it be renamed as WorkflowOperator to make it clear?

Good suggestion.

Reply:
> If possible, I would prefer an alternative usage like `NewApplicationClient(k8sClient).Workflow().Suspend(ctx, app)` but it is okay for the WorkflowOperator as well.

Since the client only update workflow currently, feel free refactor a this in the future?

Reply:
Absolutely ok.